pkg/controller/build/build_controller.go

```diff
@@ -324,19 +324,11 @@ func (ctrl *Controller) processNextMosWorkItem() bool {
 
 // Reconciles the MachineConfigPool state with the state of a custom pod object.
 func (ctrl *Controller) customBuildPodUpdater(pod *corev1.Pod) error {
-	pool, err := ctrl.mcfgclient.MachineconfigurationV1().MachineConfigPools().Get(context.TODO(), pod.Labels[constants.TargetMachineConfigPoolLabelKey], metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-
 	klog.V(4).Infof("Build pod (%s) is %s", pod.Name, pod.Status.Phase)
 
-	mosc, mosb, err := ctrl.getConfigAndBuildForPool(pool)
-	if err != nil {
-		return err
-	}
-	if mosc == nil || mosb == nil {
-		return fmt.Errorf("Missing MOSC/MOSB for pool %s", pool.Name)
+	mosc, mosb, err := ctrl.getConfigAndBuildFromBuildPodAnnotations(pod)
+	if err != nil || mosc == nil || mosb == nil {
+		return fmt.Errorf("Missing MOSC/MOSB for pod:%s due to:%v", pod.Name, err)
```

Comment:
If `err == nil && mosc == nil`, the output might be confusing. How about:
```suggestion
	if err != nil {
		return fmt.Errorf("Missing MOSC/MOSB for pod:%s due to:%v", pod.Name, err)
	if mosc == nil || mosb == nil {
		return fmt.Errorf("Missing MOSC/MOSB for pod:%s", pod.Name)
```